lib/storage/src/content_manager/consensus/persistent.rs

```diff
@@ -126,10 +130,25 @@ impl Persistent {
             state
         };
 
+        if state.remove_unknown_peer_metadata() {
+            state.save()?;
+        }
+
         log::debug!("State: {:?}", state);
         Ok(state)
     }
 
+    fn remove_unknown_peer_metadata(&self) -> bool {
+        let mut peer_metadata = self.peer_metadata_by_id.write();
+        let peer_metadata_len = peer_metadata.len();
+
+        let peer_address = self.peer_address_by_id.read();
+        peer_metadata.retain(|peer_id, _| peer_address.contains_key(peer_id));
+
+        // Return `true`, if `peer_metadata` was updated
+        peer_metadata_len != peer_metadata.len()
+    }
```

Comment:
How about moving the save into the `remove_unknown_peer_metadata` itself?

Reply:
Done.